File: 프로그래머스/0/120848. 팩토리얼/팩토리얼.cs
using System;

public class Solution 
{
    public int solution(int n) 
    {
        int i = 1;
        int count = 1;
        while(i <= n)
        {
            i *= (count + 1);
            count++;
        }
        return count - 1;
    }
}

File: 프로그래머스/0/120848. 팩토리얼/팩토리얼.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) 
{
    int i=1;
    int c = 1;
    while(true)
    {
        if(i == n)
        {
            break;
        }
        else if(i > n)
        {
            c--;
            break;
        }
        else
        { 
            c++;
            i *= c;
        }
    }
    return c;
}